python/lsst/analysis/ap/nb_utils.py

```diff
@@ -70,3 +77,222 @@ def make_simbad_link(ra, dec, radius_arcsec=3.0):
         print(f"No matched sources within {radius_arcsec} arcseconds.")
 
         return None
+
+
+def compare_sources(butler1, butler2, query1, query2,
+                    diffimg_type1='goodSeeingDiff',
+                    diffimg_type2='goodSeeingDiff',
+                    badFlagList=None, matchRadius=0.1,
+                    makeCutouts=False, displayCutouts=False,
+                    cutoutPath1=None, cutoutPath2=None,
+                    chunk_size1=None, chunk_size2=None,
+                    add_cutout_metadata=True, njobs=1):
+    """ compare two APDB datasets by extracting unassociated sources,
+    spatially crossmatching, and plotting cutouts of the differences.
+
+    Parameters
+    ----------
+
+    butler1: `lsst.daf.butler`
```

Comment:
There must be a space after the variable name and before the colon: https://developer.lsst.io/python/numpydoc.html#returns